### PR TITLE
docs: run doctoc and create type docs

### PR DESCRIPTION
**Pull request by joshblack**
Add docs for `@carbon/type`

#### Changelog

**New**

**Changed**

- `@carbon/type`
  - Update README.md
  - Remove `bundle.scss`

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
cc @jendowns @vpicone if you have a chance to review! 😄 🙏

File: packages/type/README.md [hunk 1]
```diff
@@ -25,15 +25,250 @@ yarn add @carbon/type
...
+| Feature                       | Description                                                                                                  |
+| ----------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| [Font face](#font-face)       | Include IBM Plex™ font faces in your application. Uses Google fonts                                          |
+| [Type classes](#type-classes) | Helpers to use type styles directly. Not included by default                                                 |
+| [Font family](#font-family)   | Defines the font stack for IBM Plex™ in your application, provides helpers for working with font definitions |
+| [Reset](#reset)               | Provides a high-level CSS Reset to use in your project                                                       |
+| [Scale](#type-scale)          | Provides the type scale. Can access the size at any given step                                               |
+| [Styles](#type-styles)        | Provides type styles for your application                                                                    |
```
Comment (jendowns) [inline packages/type/README.md]: Two questions here --
* Would it make sense to include a couple examples of what styles would be included in parentheses? (personally I got mixed up between styles and scale earlier this week)
* Can styles include scale? (But not the other way around of course)

File: packages/type/README.md [hunk 2]
```diff
@@ -25,15 +25,250 @@ yarn add @carbon/type
...
+```scss
+@import '@carbon/type/scss/classes';
+```
+
+| Selector                 | Description                             |
+| ------------------------ | --------------------------------------- |
+| `.bx--type-mono`         | Specify the font face as IBM Plex Mono  |
+| `.bx--type-sans`         | Specify the font face as IBM Plex Sans  |
+| `.bx--type-serif`        | Specify the font face as IBM Plex Serif |
+| `.bx--type-light`        | Specify the font weight as light        |
```
Comment (jendowns) [inline packages/type/README.md]: For these do you think it would be useful to also include the number weight for quick ref?

**Comment by vpicone** [comment]
It might be worth exploring a different option for documenting the elements urls. Maybe combining them into one section or as a comment in the `@import` code blocks. My concern is that it might start to look cluttered if we give the alternate import url after each subsection in all the packages.

File: packages/type/README.md (hunk 1 above)
Comment (joshblack) [reply packages/type/README.md]: > * So would styles include scale? (But not the other way around of course)

Yup!

**Comment by joshblack** [comment]
Thanks @vpicone and @jendowns! Updated based on feedback 😄

**Comment by jendowns** [comment]
Looks great @joshblack 🎉

**Pull request merged**